### DISCUSSION TITLE
How to generate under Qt 5.15.2 and 6?

### DISCUSSION THREAD

**Discussion by Sepera-okeq** [MeVisLab/pythonqt#170] [Q&A] [answered]
How to generate code for Qt 5.15.2 and 6? 
My generator is not working properly:
```
Please wait while source files are being generated...
Parsing typesystem file [:/trolltech/generator/build_all.txt]
Fatal error: line=1, column=1, message=unexpected end of file

Fatal error: line=2, column=62, message=Failed to parse: 'typesystem_core.xml'

Cannot parse file: ':/trolltech/generator/build_all.txt'
```
How to fix it, is it a problem of this project? And how to create your own configuration (.xml) files?

And another question. I would like to control which functions can be used in Python.  
That is, the ability to cut out functions so that they cannot be used. Is it possible with this generator?

**Comment by iakov** [comment, marked as answer]
1) Set QTDIR to some dir. It is better to create a folder like `qt5dir_for_generator` and symlink `/usr/include/x86_64-linux-gnu/qt5/include` into it. Then `export QTDIR=....`.
2) run `pythonqt_generator` from the `generator` folder, where those `*.xml`s are located
3) use `../generated_cpp`

**Comment by iakov** [comment]
Update. Qt5.15 is now supported, Qt6-related files can be generated as was previously suggested above. Worth noting, that `qmake6` should be used (on most distributions) to force Qt6 instead of Qt5.
